base/src/main/java/org/mozilla/jss/ssl/javax/JSSSession.java

```diff
@@ -64,8 +69,12 @@ public SSLChannelInfo getChannelInfo() {
     }
 
     public SSLPreliminaryChannelInfo getPreliminaryChannelInfo() {
-        if (parent.getSSLFDProxy() != null) {
-            return SSL.GetPreliminaryChannelInfo(parent.getSSLFDProxy());
+        JSSEngine jssEngine = parent.get();
+        if (jssEngine == null) {
+          return null;
+        }
+        if (jssEngine.getSSLFDProxy() != null) {
+            return SSL.GetPreliminaryChannelInfo(jssEngine.getSSLFDProxy());
```

Comment:
The return value of `jssEngine.getSSLFDProxy()` probably should be stored in a variable so it can be reused.

Reply:
Saved as ssl_fd.  I would have used sslfdProxy, but the code already uses ssl_fd everywhere else.